go/gilded-rose_test.go

```diff
@@ -1,15 +1,108 @@
 package main
 
-import "testing"
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
 
-func Test_Foo(t *testing.T) {
-	var items = []*Item{
-		&Item{"foo", 0, 0},
+func TestUpdateQuality(t *testing.T) {
+	var tests = []struct {
+		testname string
+		in       []*Item
+		want     []*Item
+	}{
+		{testname: "A 1 1", in: []*Item{{name: "A", sellIn: 1, quality: 1}}, want: []*Item{{name: "A", sellIn: 0, quality: 0}}},
```

Comment:
The test name is not informative, I should be able to understand what the purpose of the test by looking at the name.
Example for general good names:
"Sanity success test"
"Fail on negative quality"
"Hebrew item name"